docs/source/transaction_family_specifications/track_and_trade_transaction_family.rst

```diff
@@ -0,0 +1,717 @@
+           INT = 2;
+           FLOAT = 3;
+           LOCATION = 4;
+       }
+
+       // The name of the property, e.g. `temperature`
+       string name = 1;
+       DataType data_type = 2;
+       bool required = 3;
+   }
+
+   message RecordType {
+       // A unique human-readable designation for the RecordType
+       string name = 1;
+
+       repeated PropertySchema properties = 2;
+   }
+
+
+A Record will have all and only the Properties listed in its type, and
+new Records cannot be created without a type. As a consequence, a
+type-creation transaction must be executed before any Records can be
+created.
+
+RecordTypes whose addresses collide will be stored in a list
+alphabetized by name.
+
+.. code-block:: protobuf
+
+   message RecordTypeContainer {
+       repeated RecordType entries = 1;
+   }
+
+
+Because it is expected to be used for many RecordTypes, a dedicated
+Location protobuf message will be used, the values of which will be
+latitude and longitude.
+
+.. code-block:: protobuf
+
+  message Location {
+        // Coordinates are expected to be in millionths of a degree
+        sint64 latitude = 1;
+        sint64 longitude = 2;
+  }
+
+
+Agents
+------
+
+Agents are entities that can send transactions affecting Records. This
+could include not only humans and companies that act as owners and
+custodians of objects being tracked, but also autonomous sensors
+sending transactions updating Records' data. All Agents must be
+registered on-chain before interacting with Records.
+
+.. code-block:: protobuf
+
+    message Agent {
+        // The Agent's public key. This must be unique.
+        string public_key = 1;
+
+        // A human-readable name identifying the Agent.
+        string name = 2;
+
+        // Approximately when the Agent was registered, as a Unix UTC timestamp
+        uint64 timestamp = 3;
+    }
+
+Agents whose keys have the same hash will be stored in a list
+alphabetized by public key.
+
+.. code-block:: protobuf
+
+    message AgentContainer {
+        repeated Agent entries = 1;
+    }
+
+
+Proposals
+---------
+
+A Proposal is an offer from the owner or custodian of a Record to
+authorize another Agent as an owner, custodian, or reporter for that
+Record. Proposals are tagged as being for transfer of ownership,
+transfer of custodianship, or authorization of a reporter for some
+Properties. Proposals are also tagged as being open, accepted,
+rejected, or cancelled. For a given issuing Agent, a Record cannot
+have more than one open Proposal for transfer of ownership per
+receiving Agent, more than one open Proposal for transfer of
+custodianship per receiving Agent, or more than one open reporter
+Proposal per reporter.
+
+.. code-block:: protobuf
+
+   message Proposal {
+       enum Role {
+           OWNER = 1;
+           CUSTODIAN = 2;
+           REPORTER = 3;
+       }
+
+       enum Status {
+           OPEN = 1;
+           ACCEPTED = 2;
+           REJECTED = 3;
+           CANCELLED = 4;
```

Comment:
Is this how it is spelled in the `proposal.protos`? We should fix the spelling there as well.

Reply:
https://en.wikipedia.org/wiki/American_and_British_English_spelling_differences#Doubled_consonants

I can change it if you like, but it is correct for some standard of correctness.

Reply:
Me, I prefer British spelling throughout, but that means "colour," "license," "recognise," and "manoeuvre". (And lots of spell checking.) It's probably easier to use the standard American spelling.

Reply:
I think we should stick to one standard of correctness.